File: 알고리즘/2/Hash/jewels and stones/Jewels and Stones.cpp
class Solution
{
public:
    int numJewelsInStones(string jewels, string stones)
    {
        unordered_map<char, int> map;

        for (auto ele : stones)
        {
            map[ele] += 1;
        }

        int cnt = 0;
        for (auto ele : jewels)
        {
            cnt += map[ele];
        }

        return cnt;
    }
};

File: algorithms/2/Hash/jewels and stones/Jewels and Stones.py
from collections import Counter

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        return sum([ele in jewels for ele in stones])

    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        dict1 = Counter(stones)
        cnt = 0
        for ele in jewels:
            cnt+=dict1[ele]        
        return cnt